src/Raven.Server/ServerWide/ClusterStateMachine.cs

```diff
@@ -418,6 +418,8 @@ private void RemoveNodeFromCluster(TransactionOperationContext context, Blittabl
                     if (record.Topology.RelevantFor(removed))
                     {
                         record.Topology.RemoveFromTopology(removed);
+                        // Explict removing of the node means that we modify the replication factor
+                        record.Topology.ReplicationFactor = record.Topology.Count;
```

Comment:
We also need to check for other locations, where we manually add / remove items to the topology, no?